v-next/hardhat/src/internal/cli/telemetry/analytics/utils.ts

```diff
@@ -0,0 +1,90 @@
+import type { AnalyticsFile } from "./types.js";
+
+import os from "node:os";
+import path from "node:path";
+
+import { getTelemetryDir } from "@ignored/hardhat-vnext-core/global-dir";
+import { isCi } from "@ignored/hardhat-vnext-utils/ci";
+import {
+  exists,
+  readJsonFile,
+  writeJsonFile,
+} from "@ignored/hardhat-vnext-utils/fs";
+
+const ANALYTICS_FILE_NAME = "analytics.json";
+
+export async function getClientId(): Promise<string> {
+  let clientId = await readAnalyticsId();
+
+  if (clientId === undefined) {
+    // If the clientId is not undefined and it is of type "firstLegacy" or "secondLegacy," store it in the new file format
+    clientId =
+      (await readSecondLegacyAnalyticsId()) ??
+      (await readFirstLegacyAnalyticsId());
+
+    if (clientId === undefined) {
+      const { v4: uuid } = await import("uuid");
+      // TODO:log log("Client Id not found, generating a new one");
+      clientId = uuid();
+    }
+
+    await writeAnalyticsId(clientId);
+  }
+
+  return clientId;
+}
+
+export function getUserType(): string {
+  return isCi() ? "CI" : "Developer";
```

Comment:
We don't send analytics hits from CIs anymore, so we should remove the userType

Reply:
[Done here](https://github.com/NomicFoundation/hardhat/pull/5545/commits/c01024a1eb3e43e07cfd6fcfc4cb8fc8d9b19dc2)